rpcs3/Emu/Io/LogitechG27.cpp

```diff
@@ -88,12 +88,14 @@ static void clear_sdl_joysticks(std::map<u32, std::vector<SDL_Joystick*>>& joyst
 
 usb_device_logitech_g27::~usb_device_logitech_g27()
 {
-	// stop the house keeping thread
-	m_stop_thread = true;
-
 	// wait for the house keeping thread to finish
-	if (m_house_keeping_thread.joinable())
-		m_house_keeping_thread.join();
+	if (m_house_keeping_thread)
+	{
+		auto& thread = *m_house_keeping_thread;
+		thread = thread_state::aborting;
+		thread();
```

Comment:
Technucally those two lines are not needed, they are implocit in the destructor of named_thread

Reply:
Just copy pasting the same code for the 20th time

Reply:
It turned out to be a UB, because the reference existed after destruction.